### PR TITLE
feature/8502-chris-appointments-list-view-updates

### PR DESCRIPTION
**Pull request by alexandec**
## Description of Change

Update the appointments list view to standardize appointment display, and simplify/clean up the code.

Based on the `feature-xl/AppointmentsComponentization` branch which will contain all our Appointments improvements/refactoring.

## Screenshots/Video

Before/after:
&nbsp;&nbsp;

## Testing

- [x] Tested on iOS 
- [x] Tested on Android 

## PR Checklist

  **Reviewer:** Confirm the items below as you review

- [ ] PR is connected to issue(s)
- [ ] Tests are included to cover this change (when possible)
- [ ] No magic strings (All string unions follow the [Union -> Constant](https://github.com/department-of-veterans-affairs/va-mobile-app/blob/develop/VAMobile/src/constants/common.ts) type pattern)
- [ ] No secrets or API keys are checked in
- [ ] All imports are absolute (no relative imports)
- [ ] New functions and Redux work have proper TSDoc annotations

## For QA

[Run a build for this branch](https://github.com/department-of-veterans-affairs/va-mobile-app/actions/workflows/on_demand_build.yml)

### CODE REVIEW

File: VAMobile/e2e/tests/AppointmentsExpanded.e2e.ts [hunk 1]
```diff
@@ -20,43 +18,47 @@ export const Appointmentse2eConstants = {
...
+  } else if (appointmentType === 'VA' && !pastAppointment) {
     await expect(element(by.text('In-person appointment'))).toExist()
-  } else if (appointmentType == 'Claim') {
+  } else if (appointmentType === 'Claim') {
     await expect(element(by.text('Claim exam'))).toExist()
     await expect(element(by.id('directionsTestID'))).toExist()
     await expect(element(by.id('CallVATestID'))).toExist()
     await expect(element(by.id('CallTTYTestID'))).toExist()
-  } else if (appointmentType == 'Covid') {
+  } else if (appointmentType === 'Covid') {
```
Comment (Sparowhawk) [inline VAMobile/e2e/tests/AppointmentsExpanded.e2e.ts:61]: Per Misty

Misty Milliron-Grant
  4 days ago
Check in with web on this, 
@Dylan Nienberg
. COVID-19 vaccine appt should be handled exactly like VA in-person appts.

These are no longer a separate type and now fall back into the VA type

File: VAMobile/src/screens/HealthScreen/Appointments/AppointmentDetailsCommon/PreferredAppointmentType.tsx [hunk 2]
```diff
@@ -30,7 +30,7 @@ function PreferredAppointmentType({ attributes }: PreferredAppointmentTypeProps)
       case AppointmentTypeConstants.VA_VIDEO_CONNECT_HOME:
       case AppointmentTypeConstants.VA_VIDEO_CONNECT_ONSITE:
       case AppointmentTypeConstants.VA_VIDEO_CONNECT_GFE:
-        preferredTypeofAppointment = t('appointments.pending.video')
+        preferredTypeofAppointment = t('video')
```
Comment (Sparowhawk) [inline VAMobile/src/screens/HealthScreen/Appointments/AppointmentDetailsCommon/PreferredAppointmentType.tsx:33]: if mine gets merged first then there will be a conflict with this string, I didn't change any names in my branch I believe

Comment (Sparowhawk) [reply VAMobile/src/screens/HealthScreen/Appointments/AppointmentDetailsCommon/PreferredAppointmentType.tsx:33]: just calling it out so we don't forget once either is merged

File: VAMobile/src/utils/appointments.tsx [hunk 3]
```diff
@@ -280,35 +297,76 @@ const getListItemsForAppointments = (
...
+  text: getFormattedTimeForTimeZone(startDateUtc, timeZone),
+  variant: 'MobileBodyBold',
+  mb,
+})
+
+const getCareText = (typeOfCare: string, serviceCategoryName: string, isCovidVaccine: boolean, t: TFunction) => {
+  if (serviceCategoryName === 'COMPENSATION & PENSION') {
+    return t('appointments.claimExam')
+  } else if (isCovidVaccine) {
+    return t('upcomingAppointments.covidVaccine')
```
Comment (Sparowhawk) [inline VAMobile/src/utils/appointments.tsx:344]: covid vaccines again are no longer a separate type

**Review by Sparowhawk — CHANGES_REQUESTED** [review]
Everything looks good aside from covid vaccine's are no longer a separate type per slack discussion with Misty and Web is no longer displaying them as such. 

Also noted that there may be some string syncing issues between our two branches that we will need to fix, but that is the only merge conflicts that I can see between the two.

File: VAMobile/e2e/tests/AppointmentsExpanded.e2e.ts (hunk 1 above)
Comment (alexandec) [reply VAMobile/e2e/tests/AppointmentsExpanded.e2e.ts]: Thanks, I removed the covid-specific E2E tests to match your PR

File: VAMobile/src/utils/appointments.tsx (hunk 3 above)
Comment (alexandec) [reply VAMobile/src/utils/appointments.tsx]: Removed the isCovidVaccine case here

File: VAMobile/src/screens/HealthScreen/Appointments/AppointmentDetailsCommon/PreferredAppointmentType.tsx (hunk 2 above)
Comment (alexandec) [reply VAMobile/src/screens/HealthScreen/Appointments/AppointmentDetailsCommon/PreferredAppointmentType.tsx:33]: Makes sense, shouldn't be a hard conflict to fix

**Review by rbontrager — APPROVED** [review]

**Pull request merged**